### ISSUE TITLE
Bg for NOTE has bad contrast with fg

### ISSUE THREAD

**Issue by extremq** [AstroNvim/astrotheme#142]
### Checklist

- [X] I have searched through the AstroNvim documentation
- [X] I have searched through the existing issues of this project
- [X] I have searched the existing issues of plugins related to this issue
- [X] I can replicate the bug with the minimal `repro.lua` provided below

### Neovim version (nvim -v)

0.10.0

### Operating system/version

ubuntu

### Terminal/GUI

windows terminal

### Describe the bug

![image](https://github.com/user-attachments/assets/077ada0a-88f2-427a-ae43-3ff031d5a830)
This is the astronvim default theme and this is how NOTE comments look. The contrast is unreadable.

### Steps to Reproduce

Just do a -- NOTE in a .lua file with default astronvim config.

### Expected behavior

The contrast should be better, like this 
![image](https://github.com/user-attachments/assets/60b02495-b8a9-415a-927d-e1d0bf3037f3)

### Additional Context

I have fixed it for me by overriding the highlights dictionary
```lua
highlights = {
      init = { -- this table overrides highlights in all themes
        -- Normal = { bg = "#000000" },
      },
      astrodark = { -- a table of overrides/changes when applying the astrotheme theme
        TodoBgNOTE = {
          fg = "#1A1D23",
          bg = "#00B298",
        },
        -- Normal = { bg = "#000000" },
      },
    },
```

### Minimal configuration

```Lua
-- save as repro.lua
-- run with nvim -u repro.lua
-- DO NOT change the paths
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "runtime", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  -- stylua: ignore
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", "--branch=stable", lazypath })
end
vim.opt.rtp:prepend(vim.env.LAZY or lazypath)

-- install plugins
local plugins = {
  { "AstroNvim/AstroNvim", import = "astronvim.plugins" },
  { "AstroNvim/astrocommunity" },

  -- add any other plugins/customizations here
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

-- add anything else here (autocommands, vim.filetype, etc.)
```

**Comment by A-Lamia** [comment]
Hey!

I think this was a side effect from a time you could not change the foreground as the plugin would override the foreground with it's own automatic contrast function, it seems now it's not the case anymore so let me just give it a look over.

Thanks.

**Comment by mehalter** [comment]
I think it's technically not a public API still and not a real supported feature. I think depending on how stuff is configure the highlight groups that get created have different names. So changing just the TODO would not fix every case. This might need to be an issue upstream about fixing contrast detection. I think if we were to fix it here, we would need to change the color of the blue

**Comment by mehalter** [comment]
Yeah, I checked back to the source code and the proposed solution is not valid because the highlight group names are not stable nor "public facing". It is dictated by the `opts` table. I looked into the code for `todo-comments.nvim` and submitted a fix: https://github.com/folke/todo-comments.nvim/pull/316


Thanks! I'll go ahead and close this and mark it as an upstream issue. Hopefully it gets fixed very soon :)
